src/Desktop.Analyzers/Core/DoNotUseInsecureDTDProcessing.cs

```diff
@@ -0,0 +1,860 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Desktop.Analyzers.Common;
+using Analyzer.Utilities;
+
+namespace Desktop.Analyzers
+{
+    /// <summary>
+    /// Secure DTD processing and entity resolution in XML
+    /// </summary>
+    public abstract class DoNotUseInsecureDTDProcessingAnalyzer<TLanguageKindEnum> : DiagnosticAnalyzer where TLanguageKindEnum : struct
+    {
+        internal const string RuleId = "CA3075";
+        private const string HelpLink = "http://aka.ms/CA3075";
+        // Do not use insecure API:
+
+        // Do not use overloads that enable dtd processing, use one takes XmlReader instead.
+        // Matches FxCop warnings 3056,3057,3059, 3060, 3062, 3063, 3064, 3070, 3071, 3072
+        internal static DiagnosticDescriptor RuleDoNotUseUnsafeDtdProcessingOverloads = CreateDiagnosticDescriptor(SecurityDiagnosticHelpers.GetLocalizableResourceString(nameof(DesktopAnalyzersResources.DoNotUseDtdProcessingOverloadsMessage)),
```

Comment:
I am good with the first part. Not agree with the second one. I am still using the helper to create the remaining one rule, by which avoids a big chunk of code in a field declaration.